config-service-impl/src/test/java/org/hypertrace/config/service/store/DocumentConfigStoreTest.java

```diff
@@ -76,42 +81,51 @@ void writeConfig() throws IOException {
     Document document = documentCaptor.getValue();
     long newVersion = CONFIG_VERSION + 1;
     assertEquals(new ConfigDocumentKey(configResource, newVersion), key);
-    assertEquals(getConfigDocument(DEFAULT_CONTEXT, newVersion, config1), document);
+    assertEquals(getConfigDocument(DEFAULT_CONTEXT, newVersion, config1, TIMESTAMP1,
+        ((ConfigDocument) document).getUpdateTimestamp()), document);
   }
 
   @Test
   void getConfig() throws IOException {
```

Comment:
Can we add a backwards compatibility test in here? That is, a document without a create/update timestamp

Reply:
Added a backwards compatibility test for `ConfigDocument`. Here, I think it won't make much sense because we would simply be returning a mocked document with creation and update timestamp as 0 in the mocked call.